File: 1667-find-kth-bit-in-nth-binary-string/1667-find-kth-bit-in-nth-binary-string.py
class Solution:
    def findKthBit(self, n: int, k: int) -> str:
        s = "0"
        for _ in range(1,n):
            inverted = ""
            for ch in s:
                inverted += "1" if ch == "0" else "0"
            new_s = s + "1" + inverted[::-1]
            s = new_s
        # print(s)
        return s[k-1]

File: 1667-find-kth-bit-in-nth-binary-string/1667-find-kth-bit-in-nth-binary-string.java
class Solution {
     public char findKthBit(int n, int k) {
        StringBuilder str = new StringBuilder("0");
        for(int i = 0; i<n-1; i++){
            StringBuilder inverted = new StringBuilder();
            for(int j = 0; j<str.length(); j++){
                if(str.charAt(j) == '0')
                    inverted.append('1');
                else
                    inverted.append('0');
            }
            str = str.append('1').append(inverted.reverse());
        }
        return str.charAt(k-1);
    }
}